data-prepper-plugins/opensearch/src/integrationTest/java/com/amazon/dataprepper/plugins/sink/opensearch/OpenSearchSecurityAccessor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.sink.opensearch;
+
+import org.opensearch.client.Request;
+import org.opensearch.client.Response;
+import org.opensearch.client.RestClient;
+import org.opensearch.common.Strings;
+import org.opensearch.common.xcontent.XContentFactory;
+
+import javax.ws.rs.HttpMethod;
+import java.io.IOException;
+
+class OpenSearchSecurityAccessor {
+    private static final String PLUGINS_SECURITY_API = "_opendistro/_security/api/";
+    private RestClient client;
+
+    OpenSearchSecurityAccessor(RestClient client) {
+        this.client = client;
+    }
+
+    void createBulkWritingRole(String role, String indexPattern) throws IOException {
+        createRole(role, indexPattern, "indices:data/write/index", "indices:data/write/bulk*");
+    }
+
+    private void createRole(String role, String indexPattern, String... allowedActions) throws IOException {
```

Comment:
Let's mark our parameters as final so no one can accidentally declare another local final variable in the method.

We would probably catch it in code review but this will help enforce that for us.